tests/useSessionStorage.test.ts

```diff
@@ -0,0 +1,299 @@
+    const { result: r2, rerender: rerender2 } = renderHook(() => useSessionStorage('foo'));
+
+    const [, setFoo] = r1.current;
+    act(() => setFoo('potato'));
+    rerender1();
+    rerender2();
+
+    const [val1] = r1.current;
+    const [val2] = r2.current;
+
+    expect(val1).toEqual(val2);
+    expect(val1).toEqual('potato');
+    expect(val2).toEqual('potato');
+  });
+  */
+
+  it('parses out objects from sessionStorage', () => {
+    sessionStorage.setItem('foo', JSON.stringify({ ok: true }));
+    const { result } = renderHook(() => useSessionStorage<{ ok: boolean }>('foo'));
+    const [foo] = result.current;
+    expect(foo!.ok).toEqual(true);
+  });
+
+  it('safely initializes objects to sessionStorage', () => {
+    const { result } = renderHook(() => useSessionStorage<{ ok: boolean }>('foo', { ok: true }));
+    const [foo] = result.current;
+    expect(foo!.ok).toEqual(true);
+  });
+
+  it('safely sets objects to sessionStorage', () => {
+    const { result, rerender } = renderHook(() => useSessionStorage<{ ok: any }>('foo', { ok: true }));
+
+    const [, setFoo] = result.current;
+    act(() => setFoo({ ok: 'bar' }));
+    rerender();
+
+    const [foo] = result.current;
+    expect(foo!.ok).toEqual('bar');
+  });
+
+  it('safely returns objects from updates', () => {
+    const { result, rerender } = renderHook(() => useSessionStorage<{ ok: any }>('foo', { ok: true }));
+
+    const [, setFoo] = result.current;
+    act(() => setFoo({ ok: 'bar' }));
+    rerender();
+
+    const [foo] = result.current;
+    expect(foo).toBeInstanceOf(Object);
+    expect(foo!.ok).toEqual('bar');
+  });
+
+  it('sets sessionStorage from the function updater', () => {
+    const { result, rerender } = renderHook(() =>
+      useSessionStorage<{ foo: string; fizz?: string }>('foo', { foo: 'bar' })
+    );
+
+    const [, setFoo] = result.current;
+    act(() => setFoo((state) => ({ ...state!, fizz: 'buzz' })));
+    rerender();
+
+    const [value] = result.current;
+    expect(value!.foo).toEqual('bar');
+    expect(value!.fizz).toEqual('buzz');
+  });
+
+  it("function updater doesn't get stale state", () => {
+    const { result, rerender } = renderHook(() => useSessionStorage('foo', false));
+
+    act(() => result.current[1]((state) => !state));
+    rerender();
+    expect(result.current[0]).toEqual(true);
+
+    act(() => result.current[1]((state) => !state));
+    rerender();
+    expect(result.current[0]).toEqual(false);
+  });
+
+  it('rejects nullish or undefined keys', () => {
+    const { result } = renderHook(() => useSessionStorage(null as any));
+    try {
+      (() => {
+        return result.current;
+      })();
+      fail('hook should have thrown');
+    } catch (e) {
+      expect(String(e)).toMatch(/key may not be/i);
+    }
+  });
+
+  describe('remove', () => {
+    it('removes value from sessionStorage and restores initialValue', () => {
```

Comment:
I think we need to distinguish between `remove` and `restore` here.

When the consumer calls the remove function:

1. Does the user wants to delete the session key from the storage?
2. Does the user wants to restore the session key to it's initial value?

I do not have answer for it therefore I wouldn't introduce this new feature yet.